### ISSUE TITLE
Lab 9 complicated

### ISSUE THREAD

**Issue by Devvynneo** [nus-cs2030/2122-s1#259]
Is it me or... Lab 9 is confusing as... very confusing

**Comment by Albunist** [comment]
I agree that its very confusing at the start but what I did was to refer to the lab slides which was helpful as it summarized the methods that you need to change. Essentially you need to change the code by wrapping variables in a CompletableFuture with the relevant methods such as thenApply and thenCombine etc. Reading the documentation of these methods will also help to understand what they do and how they are used for the lab assignment.

**Comment by yuyanjulia** [comment]
Most of the original code is not required to change. The instruction has already specified which method to change into asynchronised (read the question carefully for hints). The return type of such method should always be a CompletableFuture of the original return type. If you are a bit unsure what to do, can try `jar uf level1.jar *.class` first and see what the error messages are. Only call `.join()` in the last level.

**Comment by callmeqimin** [comment]
just read through you will realised  that computablefuture is like optional interface and the whole lab is just about how you update the program which is actually very easy
